api/handlers/app_test.go

```diff
@@ -525,6 +530,133 @@ var _ = Describe("App", func() {
 		})
 	})
 
+	Describe("GET /v3/apps/{guid}/manifest", func() {
+		var dropletRecord repositories.DropletRecord
+		BeforeEach(func() {
+			processRecord := repositories.ProcessRecord{
+				GUID:             appGUID,
+				SpaceGUID:        spaceGUID,
+				Command:          "some-command",
+				AppGUID:          appGUID,
+				Type:             "web",
+				DesiredInstances: 1,
+				MemoryMB:         512,
+				DiskQuotaMB:      1024,
+				HealthCheck: repositories.HealthCheck{
+					Type: "http",
+					Data: repositories.HealthCheckData{
+						HTTPEndpoint:             "health-endpoint",
+						InvocationTimeoutSeconds: 5,
+						TimeoutSeconds:           10,
+					},
+				},
+			}
+			routeRecord := repositories.RouteRecord{
+				GUID:      appGUID,
+				SpaceGUID: spaceGUID,
+				Domain: repositories.DomainRecord{
+					Name: appName,
+					GUID: appGUID,
+				},
+				Host:     "host",
+				Path:     "domain",
+				Protocol: "http",
+			}
+			serviceBindingRecord := repositories.ServiceBindingRecord{
+				Name:                tools.PtrTo("binding1"),
+				ServiceInstanceGUID: "instance-guid",
+			}
+			dropletRecord = repositories.DropletRecord{
+				Lifecycle: repositories.Lifecycle{
+					Type: "docker",
+				},
+				Image: "docker-image",
+			}
+			appsStateCollector.CollectStateReturns(manifest.AppState{
+				App:       appRecord,
+				Processes: map[string]repositories.ProcessRecord{"web": processRecord},
+				Routes: map[string]repositories.RouteRecord{
+					"host." + appName + "domain": routeRecord,
+				},
+				ServiceBindings: map[string]repositories.ServiceBindingRecord{
+					"binding1": serviceBindingRecord,
+				},
+				Droplet: dropletRecord,
+			}, nil)
+
+			req = createHttpRequest("GET", "/v3/apps/"+appGUID+"/manifest", nil)
+		})
+
+		It("returns the manifest as yaml", func() {
+			Expect(appRepo.GetAppCallCount()).To(Equal(1))
+			_, actualAuthInfo, actualAppGUID := appRepo.GetAppArgsForCall(0)
+			Expect(actualAuthInfo).To(Equal(authInfo))
+			Expect(actualAppGUID).To(Equal(appGUID))
+
+			Expect(appsStateCollector.CollectStateCallCount()).To(Equal(1))
```

Comment:
I would suggest extracting this group of `Expect`s into a dedicated `It`, for example `It("collects app state"...`. Collecting the app state and returning proper http response are orthogonal aspects of the handler, therefore I believe they deserve dedicated `It`s